exense-basic-commons/src/main/java/ch/exense/commons/processes/ManagedProcess.java

```diff
@@ -253,17 +253,32 @@ public synchronized void start() throws ManagedProcessException {
                 try {
                     process = builder.start();
                 } catch (IOException e) {
-                    throw new ManagedProcessException("Unable to start the process " + id, e);
+                    throw loggedManagedProcessException("Unable to start the process " + id, e);
                 }
                 logger.debug("Started managed process " + builder.command());
-            } else {
-                throw new ManagedProcessException("Unable to start the process " + id + " twice. The process has already been started.");
+            } catch (Throwable t) {
+                try {
+                    removeTempLogDirectory();
+                } finally {
+                    if(t instanceof ManagedProcessException) {
+                        throw t;
+                    } else {
+                        throw loggedManagedProcessException("Unexpected error while starting the process " + id, t);
```

Comment:
I'm not entirely sure how the tempLogDirectory and the logger relate (if at all), so if the logging would happen into the just removed directory, then this should probably be replaced by the normal, non-logging variant of the exception. Otherwise, looks good.

Reply:
Thanks for the hint. I've checked and they do not relate.